Source/Diagnostics/ReducedDiags/LoadBalanceCosts.cpp

```diff
@@ -206,7 +206,7 @@ void LoadBalanceCosts::WriteToFile (int step) const
             int ind_rank = i - m_nDataFields + 2; // index for the rank corresponding to current box
 
             // m_data --> rank --> hostname
-            ofs << m_sep << m_data_string[m_data[ind_rank]];
+            ofs << m_sep << m_data_string[(long unsigned int)(m_data[ind_rank])];
```

Comment:
```suggestion
            ofs << m_sep << m_data_string[static_cast<long unsigned int>(m_data[ind_rank])];
```
I would just suggest to use a C++-style cast instead of a C-style cast

Reply:
Done, thanks!